### PR TITLE
Replace withCall -> withCalls

### LINKED ISSUE #1009: Convert all withCall to withCalls<Props>
Goal:

Make everything consistent, so there is basically one way of doing things.

Background:

`withCalls` is a relative new addition (replacing `withMulti(withCall(abc), withCall(def), ...)`and actually have some additional properties for types and is slightly more future-proof when changing components. 

Generally the `Props` gets lost with `withCall` and is handled (slightly) better in the case of `withCalls<Props>` - nowhere near perfect, we really would like to have decorators at some point. (But more or less given up on that for now, the TS support is just lacking and types become an absolute nightmare)

TL;DR `withCalls` has _some_ benefits, at the very least when you add more stuff, you can just add it

Actions:

- For all `withCall`, replace it with `withCalls<Props>(...)`
- Adjust props as needed, I've noticed there are places it doesn't even get specified, e.g. https://github.com/polkadot-js/apps/blob/master/packages/ui-reactive/src/Balance.tsx#L12 - no `params` in props, but actually used

### PR DESCRIPTION
**Pull request by kwingram25**
Closes #1009 

Replace all instances of withCall with withCalls<Props>

### CODE REVIEW

File: packages/apps/src/SideBar/index.tsx [hunk 1]
```diff
@@ -190,19 +191,17 @@ class SideBar extends React.PureComponent<Props> {
...
-    //   <Menu.Item className='apps--SideBar-Item'>
-    //     <a
-    //       className='apps--SideBar-Item-NavLink'
-    //       href='https://github.com/w3f/Web3-wiki/wiki/Polkadot'
-    //     >
-    //       <Icon name='book' /> Wiki
-    //     </a>
-    //   </Menu.Item>
-    // );
+    return (
```
Comment (Stefie) [inline packages/apps/src/SideBar/index.tsx]: Could you revert this change? I think it‘s part of your PR #1020

Comment (kwingram25) [reply packages/apps/src/SideBar/index.tsx]: oops! Thank you

**Comment by jacogr** [comment]
Cool. Any major issue along the way?

File: packages/app-extrinsics/src/Balance.tsx [hunk 2]
```diff
@@ -30,4 +30,9 @@ class BalanceDisplay extends React.PureComponent<Props> {
   }
 }
 
-export default withCall('query.balances.freeBalance', { paramName: 'params' })(BalanceDisplay);
+export default withMulti(
```
Comment (jacogr) [inline packages/app-extrinsics/src/Balance.tsx]: Not sure we need a withMulti here? I am guessing some types started complaining.

File: packages/app-explorer/src/BlockQuery/index.tsx [hunk 3]
```diff
@@ -83,5 +83,5 @@ class Entry extends React.Component<Props, State> {
 
 export default withMulti(
   Entry,
-  withCall('derive.chain.bestNumber')
+  withCalls<Props>('derive.chain.bestNumber')
```
Comment (jacogr) [inline packages/app-explorer/src/BlockQuery/index.tsx]: Probably can get rid of withMulti here as well?

File: packages/app-extrinsics/src/Balance.tsx (hunk 2 above)
Comment (kwingram25) [reply packages/app-extrinsics/src/Balance.tsx]: Yes, some of the reactive components threw issues when exported as withCalls(...) rather than withCall() or withMulti() - 'params' isn't supported?

Type '{ params: string | undefined; }' is not assignable to type 'IntrinsicAttributes & Pick<Props, "className" | "style" | "children" | "label" | "callResult" | "callUpdated" | "callUpdatedAt" | "accounts_idAndIndex"> & { children?: ReactNode; }'. Property 'params' does not exist on type 'IntrinsicAttributes & Pick<Props, "className" | "style" | "children" | "label" | "callResult" | "callUpdated" | "callUpdatedAt" | "accounts_idAndIndex"> & { children?: ReactNode; }'.

**Comment by jacogr** [comment]
Yes, assumed as much. The previous iteration (just due to TS being quite horrible in HOC support) actually mostly ende up with any-ing of the props. withCalls is slightly more correct (but not perfect).

**Comment by jacogr** [comment]
So where params pop up, it actually means that is probably doesn’t appear in Props (And should really)

**Comment by kwingram25** [comment]
Resolved params problems, but I had to add callOnResult type to Nonce for one instance - I can't tell if theres a prop type somewhere that it needs specifically
